st2common/st2common/util/pack_management.py

```diff
@@ -249,6 +250,13 @@ def clone_repo(temp_dir, repo_url, verify_ssl=True, ref='master'):
         repo.git.checkout('-b', short_branch, branch)
         branch = repo.head.reference
     else:
+        # Checkout out the latest tag when branch or version is not specified.
+        if not ref:
+            branch = get_pack_version_from_url(repo_url)
```

Comment:
There is no need in new function.

Please drill deeper in code based on https://github.com/StackStorm/st2/pull/4743#discussion_r304049691

Take a look at `get_repo_url` function which is called earlier that requests pack index. If we obtain/set the version there no other changes will be needed.

1) https://github.com/StackStorm/st2/blob/c70b2dbfd311f34ea98a25f9ac0b5a4e15141f12/st2common/st2common/util/pack_management.py#L108
2) https://github.com/StackStorm/st2/blob/c70b2dbfd311f34ea98a25f9ac0b5a4e15141f12/st2common/st2common/util/pack_management.py#L360
3) https://github.com/StackStorm/st2/blob/c70b2dbfd311f34ea98a25f9ac0b5a4e15141f12/st2common/st2common/util/pack_management.py#L374-L379
4) https://github.com/StackStorm/st2/blob/c70b2dbfd311f34ea98a25f9ac0b5a4e15141f12/st2common/st2common/util/pack_management.py#L153-L155

Reply:
@armab From my understanding, when pack is installed with pack name  `st2 pack install jira`, `get_pack_from_index` is called. If pack is installed with repo URL `st2 pack install https://github.com/StackStorm-Exchange/stackstorm-jira`  `get_pack_from_index` is not called and clone_repo passes `ref=None`, at this case, pack name is unknown.  Therefore, need new function to get pack version.

Reply:
By change query pack version from index file, it fixed test failure for st2cd.st2_e2e_tests_test_quickstart, but cause tests.test_run_pack_tests_tool failed.  Need to update index file for pack `csv` version from `0.1.0` to `0.4.4`.

Reply:
> From my understanding, when pack is installed with pack name st2 pack install jira, get_pack_from_index is called. If pack is installed with repo URL st2 pack install https://github.com/StackStorm-Exchange/stackstorm-jira get_pack_from_index is not called and clone_repo passes ref=None, at this case, pack name is unknown. Therefore, need new function to get pack version.

This is correct behavior.

We want to install latest stable pack version (instead of `master`) only for packs in Exchange which we control and know its consistency.
If pack is installed via URL, git, file://, - keep the previous/default behavior (install from `master` or default repo branch).

In other words, when user installs from his git repo or local folder it's expected to use current/default branch and not try to checkout any other release from git tag.

Reply:
@jinpingh According to our discussions here and in Slack, the external exchange request you're doing shouldn't happen in git clone function. It's not appropriate place for that logic.

Following my comment that stable version should be utilized on pack installations relying on StackStorm Exchange index (eg. `st2 pack install csv`) and more code pointers above, having this code: 
https://github.com/StackStorm/st2/blob/c0458f1a06c2a81cac81697817c422ffb7631047/st2common/st2common/util/pack_management.py#L360-L375

Your first and easiest change would be 
```diff
-        return (pack['repo_url'], version)
+        return (pack['repo_url'], version or pack['version'])
```

Only that will make `st2 pack install csv` working as desired and pass correct version from Exchange Index (latest stable tag).

Let's please start with that.

Reply:
@armab From my understanding, the different with my changes and your suggestion is:
For example `st2 pack install https://github.com/StackStorm-Exchange/stackstorm-jira`  or `st2 pack install jira`
My changes installed latest version of jira, that is `0.12.0`. 
```
(virtualenv) vagrant@ubuntu-xenial:/opt/stackstorm/packs/jira$ git status:
HEAD detached at v0.12.1
```
With your suggestion, it installed latest code from `master`
```
(virtualenv) vagrant@ubuntu-xenial:/opt/stackstorm/packs/jira$ git status
On branch master
```

From https://github.com/StackStorm/discussions/issues/351, requirement is `Install latest version tag, not from master.`  This is why i want to call a meeting to confirm that we are in the same page. Please let me know I misunderstand the requirement.    Thanks!